src/quo2/screens/context_tags.cljs

```diff
@@ -12,38 +12,46 @@
+(def example-pk2 "0x04c178513eb741e8c4e50326b22baefa7d60a2f4eb81e328c4bbe0b441f87b2a014a5907a419f5897fc3c0493a0ff9db689a1999d6ca7fdc63119dd1981d0c7ccf")
 
-(def descriptor [{:label "Type"
-                  :key   :type
-                  :type  :select
-                  :options [{:key :public-key
-                             :value "Public key"}
-                            {:key :avatar
-                             :value "Avatar"}
-                            {:key :group-avatar
-                             :value "Group avatar"}]}])
+(def main-descriptor [{:label   "Type"
+                       :key     :type
+                       :type    :select
+                       :options [{:key   :public-key
+                                  :value "Public key"}
+                                 {:key   :avatar
+                                  :value "Avatar"}
+                                 {:key   :group-avatar
+                                  :value "Group avatar"}]}])
 
 (defn cool-preview []
   (let [state (reagent/atom {:label "Name"
                              :type :group-avatar})]
     (fn []
-      (let [contacts @(re-frame/subscribe [:contacts/contacts])
+      (let [contacts {example-pk {:public-key example-pk
+                                  :names {:three-words-name "Automatic incompatible Coati"}}
+                      example-pk2 {:public-key example-pk2
+                                   :names {:three-words-name "Clearcut Flickering Rattlesnake"}}}
+            contacts-public-keys (map (fn [{:keys [public-key]}]
+                                        {:key   public-key
+                                         :value (multiaccounts/displayed-name
+                                                 (get contacts public-key))})
+                                      (vals contacts))
             descriptor
-            (cond-> descriptor
-              (= (:type @state) :group-avatar)
-              (conj  {:label "Label"
-                      :key :label
-                      :type :text})
-              (= (:type @state) :avatar)
-              (conj {:label "Contacts"
-                     :key :contact
-                     :type :select
-                     :options
-                     (map (fn [{:keys [public-key]}]
-                            {:key public-key
-                             :value (multiaccounts/displayed-name
-                                     (get contacts public-key))})
-                          (vals contacts))}))]
+            (cond
+              (= (:type @state) :group-avatar) (conj main-descriptor {:label "Label"
+                                                                      :key   :label
+                                                                      :type  :text})
+              (= (:type @state) :avatar) (let [photo @(re-frame.core/subscribe [:chats/photo-path (:contact @state)])]
```

Comment:
still better to not use re-frame in quo, even here, but what is the value of this subscription? can we mock it?

Reply:
I had it as a base64 string but removed it .. I'll use it again

Reply:
Done